x/precisebank/keeper/mint.go

```diff
@@ -89,8 +93,24 @@ func (k Keeper) mintExtendedCoin(
 	// x + y < (2 * ConversionFactor) - 2
 	// x + y < 1 integer amount + fractional amount
 	if newFractionalBalance.GTE(types.ConversionFactor()) {
```

Comment:
The logic looks great, however we can optimize the carry and negative remainder case to instead add to the integer amount being minted.  This avoids the extra reserve -> account transfer and the reserve mint to make up the balance change and saves quite a bit of gas.

1) No carry, zero/positive remainder -> do nothing extra
2) No carry, negative remainder -> mint to reserve
3) Carry, zero/positive remainder -> transfer 1 from reserve to account
4) Carry, negative remainder -> add 1 to integer acct mint amount